src/grpc-client/grpc_client_player.cpp

```diff
@@ -135,6 +196,28 @@ void GrpcClientPlayer::getActions()
             return;
         }
     }
+    const rcsc::WorldModel & wm = agent->world();
+
+    if ( !actions.ignore_shootinpreprocess() )
+    {
+        if ( wm.gameMode().type() != rcsc::GameMode::IndFreeKick_
+            && wm.time().stopped() == 0
+            && wm.self().isKickable()
+            && Bhv_StrictCheckShoot().execute( agent ) )
+        {
+            // reset intention
+            agent->setIntention( static_cast< rcsc::SoccerIntention * >( 0 ) );
```

Comment:
Why do we reset intention here ?

Reply:
I don't know, but isn't it better to make an exception in SoccerIntention bhv for checking the shoot possibility rather than resetting it while checking the shoot?